src/common/docker-compose/index.ts

```diff
@@ -349,15 +349,19 @@ export class DockerComposeUtils {
     }
   }
 
-  public static dockerCompose(args: string[], execa_opts?: Options, use_console = false): execa.ExecaChildProcess<string> {
+  public static dockerCompose(args: string[], execa_opts?: Options, use_console = false, on_end?: (cmd: execa.ExecaChildProcess<string>) => void): execa.ExecaChildProcess<string> {
     this.dockerCommandCheck();
     if (use_console) {
       process.stdin.setRawMode(true);
     }
     const cmd = execa('docker', ['compose', ...args], execa_opts);
     if (use_console) {
       cmd.on('exit', () => {
-        process.exit();
+        if (on_end) {
+          on_end(cmd);
+        } else {
+          process.exit();
```

Comment:
Any reason to not always use cmd.exitCode? I think I'd want that to be the default in most cases.